### DISCUSSION TITLE
`insert_many` crashes with `violates not-null constraint` when input is an empty Vec and table has null-constraints

### DISCUSSION THREAD

**Discussion by gitmalong** [SeaQL/sea-orm#1925] [General]
```rust
entities::prelude::Cake::insert_many(Vec::<crate::cake::ActiveModel>::new())
          .exec(&self.db)
          .await
          .unwrap();
```

Error

> ``` called `Result::unwrap()` on an `Err` value: Query(SqlxError(Database(PgDatabaseError { severity: Error, code: "23502", message: "null value in column \"year\" of relation \"cakes\" violates not-null constraint", detail: Some("Failing row contains (51, null, null, null, null, null, null, null, null, 0, null)."), hint: None, position: None, where: None, schema: Some("public"), table: Some("cakes"), column: Some("year"), data_type: None, constraint: None, file: Some("execMain.c"), line: Some(2003), routine: Some("ExecConstraints") })))```

Version:
```
sea-orm = { version = "0.12.3", features = ["sqlx-postgres", "runtime-tokio-rustls", "macros"] }
sea-orm-migration = "^0.12.3"
```

**Comment by gitmalong** [comment]
Is that maybe desired behavior as there is `on_empty_do_nothing` ?

**Comment by tyt2y3** [comment]
The way the query is constructed to do an `INSERT DEFAULT`, which would fill the columns with default values if defined.
So I think this is an intended behaviour.

**Reply by xamgore** [reply]
@tyt2y3 I'd like to insert instances from an iterator. `on_empty_do_nothing` converts to `INSERT VALUES(DEFAULT)` so... it does something, while I want to get a no-op in case the iterator is empty.

```rust
{
    let mut stmt = models::order_step::Entity::insert_many(path)
      .on_empty_do_nothing()
      .into_query();
    let stmt = stmt.returning_col(models::order_step::Column::Id);

    dbg!(stmt.to_string(PostgresQueryBuilder)); // <- contains INSERT VALUES(DEFAULT)

    conn
      .query_all(conn.get_database_backend().build(stmt))
      .await?
}
```

PS. Have found `do_nothing()`, though it's has the same implementation as `on_empty_do_nothing`, and is about `ON CONFLICT`.
